packages/opentelemetry-context-async-hooks/src/AsyncHooksContextManager.ts

```diff
@@ -283,6 +239,38 @@ export class AsyncHooksContextManager implements ContextManager {
    * @param uid uid of the async context
    */
   private _destroy(uid: number) {
-    this._contextRefs.delete(uid);
+    this._contexts.delete(uid);
+  }
+
+  /**
+   * Before hook is called just beforing executing a async context.
+   * @param uid uid of the async context
+   */
+  private _before(uid: number) {
+    const context = this._contexts.get(uid);
+    if (context !== undefined) {
+      this._enterContext(context);
+    }
+  }
+
+  /**
+   * After hook is called just after completing the execution of a async context.
+   */
+  private _after() {
+    this._exitContext();
+  }
+
+  /**
+   * Set the given context as active
+   */
+  private _enterContext(context: Context) {
+    this._stack.unshift(context);
```

Comment:
Why are you using shift/unshift instead of push/pop? unshift prepends on the array and requires memory allocations, where push may not. Using push here could be a significant performance improvement.

Reply:
It was to make a faster path to get the current context which would require a `.length` read instead of a direct access, but i agree that the memory impact is greater. I will change to a push/pop approach.